src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,79 @@
-export const Pagination = () => {};
+import React, { useCallback } from 'react';
+
+interface PaginationProps {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handlePageClick = useCallback((page: number) => {
+    if (page >= 1 && page <= totalPages) {
+      onPageChange(page);
+    }
+  }, [onPageChange, totalPages]);
+
+  const handlePrevClick = useCallback(() => {
+    if (currentPage > 1) {
+      onPageChange(currentPage - 1);
+    }
+  }, [currentPage, onPageChange]);
+
+  // Функція для кліку на наступну сторінку
+  const handleNextClick = useCallback(() => {
+    if (currentPage < totalPages) {
+      onPageChange(currentPage + 1);
+    }
+  }, [currentPage, onPageChange, totalPages]);
+
+  return (
+    <ul className="pagination">
+      <li className={`page-item ${currentPage === 1 ? 'disabled' : ''}`}>
+        <button
+          className="page-link"
+          data-cy="prevLink"
+          aria-disabled={currentPage === 1}
+          onClick={handlePrevClick}
+          disabled={currentPage === 1}
+        >
+          «
+        </button>
+      </li>
+
+      {Array.from({ length: totalPages }, (_, index) => (
```

Comment:
You're creating an array from an object with a `length` property to generate page numbers. This is a clever use of `Array.from`, but consider adding a comment explaining this line for future maintainability. It might not be immediately clear to other developers why you're using this approach.

Reply:
fixed